build/oss-fuzz/build.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -eu
+
+# Don't need the stdlib and libc++ gives linker error
+CXXFLAGS="${CXXFLAGS/-stdlib=libc++/}"
+
+# Needed for initialize the submodules, the submodule directories are empty.
+# Some header files (proj_api.h here) do not actually exist in cockroach, it exists in the submodule.
+git submodule update --init --recursive
+
+# Install dependencies for compile_native_go_fuzzer
+go install github.com/AdamKorcz/go-118-fuzz-build@latest
+go get github.com/AdamKorcz/go-118-fuzz-build/testing
+
+bazel clean --expunge
+
+# Generate artifacts for building the binary
+bazel run pkg/gen:code
+bazel run pkg/cmd/generate-cgo:generate-cgo --run_under "cd $SRC/cockroach && "
+
+# File is not being built via bazel and outdated (gives compilation errors in SessionIDEncoding)
+rm pkg/sql/pgwire/fuzz.go
+
+# Rename as go-118-fuzz-build cant read _test.go files (used by sessionID and frontier)
+mv ./pkg/util/span/frontier_test.go pkg/util/span/frontier_test_fuzz.go
+
+# Build the native go fuzz targets
+compile_native_go_fuzzer ./pkg/keys FuzzPrettyPrint fuzzPrettyPrint
+
+compile_native_go_fuzzer ./pkg/sql/sqlliveness/slstorage FuzzSessionIDEncoding fuzzSessionIDEncoding
+
+compile_native_go_fuzzer ./pkg/util/span FuzzBtreeFrontier fuzzBtreeFrontier
+
+# hacky fix since we've already copied these files (used in coverage only)
```

Comment:
Maybe we don't need this hack? When would we ever use coverage-only mode?

Reply:
It is meant for the code coverage report generation since it uses [different configuration](https://google.github.io/oss-fuzz/advanced-topics/code-coverage/#build-fuzz-targets) than the one that normal build uses.

Reply:
So in [`compile_native_go_fuzzer`](https://github.com/google/oss-fuzz/blob/4c5a853e1fbe1bf6a449d67f28958b6016515f8a/infra/base-images/base-builder/compile_native_go_fuzzer#L52), after`grep` we were getting 2 files with the function name in the directory(it was being referred to in a comment of another file) which was leading to `cp` to throw errors. Made the fix and no more need this hack.